src/common/m_mpi_common.fpp

```diff
@@ -51,6 +51,30 @@ contains
 
 #ifdef MFC_MPI
 
+#ifdef MFC_PRE_PROCESS
+        if (n > 0) then
+            if (p > 0) then
+
+                halo_size = -1 + buff_size*sys_size* &
+                                        & (m + 2*buff_size + 1)* &
+                                        & (n + 2*buff_size + 1)* &
+                                        & (p + 2*buff_size + 1)/ &
+                                        & (min(m, n, p) + 2*buff_size + 1)
+            else
+                halo_size = -1 + buff_size*sys_size* &
+                                        & (max(m, n) + 2*buff_size + 1)
+            end if
+        else
+            halo_size = -1 + buff_size*sys_size
+        end if
+
+        v_size = sys_size
+
+        allocate (q_prims_buff_send(0:halo_size))
+
+        allocate (q_prims_buff_recv(0:ubound(q_prims_buff_send, 1)))
+#endif
```

Comment:
The `#ifdef MFC_PRE_PROCESS` here and the `#ifdef MFC_SIMULATION` following this can be removed and `halo_size` can be used for both without any pre processor statements.